src/App.tsx

```diff
@@ -26,70 +27,117 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
   // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((goods1, goods2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return goods1.localeCompare(goods2);
+
+      case SortType.LENGTH:
+        return goods1.length - goods2.length;
+
+      default:
+        return SortType.NONE;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface State {
+  isReversed: boolean,
+  sortType: SortType,
+}
+
+export class App extends React.PureComponent<{}, State> {
+  state: Readonly<State> = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  handleSortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
```

Comment:
I don't think there's a reason to add `handle` before the name of each method
```suggestion
  sortByAlphabet = () => {
    this.setState({ sortType: SortType.ALPHABET });
```